src/landing/search_bar.rs

```diff
@@ -194,9 +207,21 @@ impl WidgetMatchEvent for SearchBar {
             }
         }
 
-        if let Some(_) = input.changed(actions) {
-            cx.stop_timer(self.search_timer);
-            self.search_timer = cx.start_timeout(self.search_debounce_time);
+        if let Some(val) = input.changed(actions) {
+            if val.is_empty() {
+                x_button.set_visible(false);
+                cx.stop_timer(self.search_timer);
+                self.search_timer = cx.start_timeout(self.search_debounce_time);
+            } else {
+                x_button.set_visible(true);
+                cx.stop_timer(self.search_timer);
+                self.search_timer = cx.start_timeout(self.search_debounce_time);
+            }
+        }
```

Comment:
We can avoid duplicating code here, you can write something like this:
```rs
        if let Some(val) = input.changed(actions) {
            if val.is_empty() {
                x_button.set_visible(false);
            } else {
                x_button.set_visible(true);
            }
            cx.stop_timer(self.search_timer); // don't write and the below these lines twice
            self.search_timer = cx.start_timeout(self.search_debounce_time);
        }
 ```
I suggest to even shorten it further:
 ```rs
 if let Some(text) = input.changed(actions) {
    x_button.set_visible(!text.is_empty());
    cx.stop_timer(self.search_timer);
    self.search_timer = cx.start_timeout(self.search_debounce_time);
}
 ```

Reply:
thank you for your suggestion